apps/src/templates/sessions/LockoutPanel.tsx

```diff
@@ -37,22 +52,50 @@ export default function LockoutPanel(props) {
 
   // When the email field is updated, also update the disability state of the
   // submit button.
-  const onEmailUpdate = event => {
-    setDisabled(!validateEmail(event.target.value));
+  const onEmailUpdate = (event: React.FormEvent<HTMLInputElement>) => {
+    setDisabled(!validateEmail((event.target as HTMLInputElement).value));
   };
 
   // This will set the email to the current pending email and fire off the
   // form as though they had typed in the same email again.
-  const resendPermissionEmail = event => {
-    const field = document.getElementById('parent-email');
-    field.value = props.pendingEmail;
+  const handleResend = () => {
+    const field = document.getElementById('parent-email') as HTMLInputElement;
+    if (field && props.pendingEmail) {
+      field.value = props.pendingEmail;
+    }
+
+    const form = document.getElementById(
+      'lockout-panel-form'
+    ) as HTMLFormElement;
+    if (form) {
+      reportEvent(EVENTS.CAP_PARENT_EMAIL_RESEND, {
+        inSection: currentUser.inSection,
+        consentStatus: currentUser.childAccountComplianceState,
+      });
+      form.submit();
+    }
+  };
+
+  const handleSubmit = () => {
+    reportEvent(EVENTS.CAP_PARENT_EMAIL_SUBMITTED, {
+      inSection: currentUser.inSection,
+      consentStatus: currentUser.childAccountComplianceState,
+    });
+  };
 
-    const form = document.getElementById('lockout-panel-form');
-    form.submit();
+  const handleUpdate = () => {
+    reportEvent(EVENTS.CAP_PARENT_EMAIL_UPDATED, {
+      inSection: currentUser.inSection,
+      consentStatus: currentUser.childAccountComplianceState,
+    });
   };
```

Comment:
There are two issues:
1. The form ignores validation errors and counts them as successes. Because of this, we will report an event even if it didn't actually happen:
https://github.com/code-dot-org/code-dot-org/blob/3aa9278211cc8e4b7e04a04aa7f2446e5272b116/dashboard/app/controllers/policy_compliance_controller.rb#L67-L70 https://github.com/code-dot-org/code-dot-org/blob/5cc0038e840532e70d2dc79d86e9bdc7863d9239/dashboard/lib/forms/child_account/parental_permission_request.rb#L16-L22
2. When users click `Update` for the same email, it's technically the `Resend` event and not the`Update`

Reply:
Please check [LockoutLinkedAccounts.jsx](https://github.com/code-dot-org/code-dot-org/pull/59382/files#diff-215ae7d0a710b6895daa1a51180e6d79b1e280760556559869c1e265ff22f794), where I had to deal with these issues as well:
https://github.com/code-dot-org/code-dot-org/blob/2811284a88f0e353cc1edf9c9b540246376c0dfd/apps/src/templates/policy_compliance/LockoutLinkedAccounts.jsx#L73-L100

Reply:
I stole a lot of code from [LockoutLinkedAccounts.jsx](https://github.com/code-dot-org/code-dot-org/pull/59382/files#diff-215ae7d0a710b6895daa1a51180e6d79b1e280760556559869c1e265ff22f794). I will create a JIRA to consolidate all of the various CAP Lockout widgets.